packages/preferences/src/browser/preference-frontend-module.ts

```diff
@@ -6,21 +6,17 @@
  */
 
 import { ContainerModule, interfaces, } from 'inversify';
-import { PreferenceProviders } from "@theia/core/lib/browser/preferences";
+import { PreferenceProvider, PreferenceScope } from "@theia/core/lib/browser/preferences";
 import { UserPreferenceProvider } from './user-preference-provider';
 import { WorkspacePreferenceProvider } from './workspace-preference-provider';
 
-export function bindPreferences(bind: interfaces.Bind, rebind: interfaces.Rebind): void {
-    bind(UserPreferenceProvider).toSelf().inSingletonScope();
-    bind(WorkspacePreferenceProvider).toSelf().inSingletonScope();
+export function bindPreferences(bind: interfaces.Bind, unbind: interfaces.Unbind): void {
+    unbind(PreferenceProvider);
 
-    rebind(PreferenceProviders).toFactory(ctx => () => {
-        const userProvider = ctx.container.get(UserPreferenceProvider);
-        const workspaceProvider = ctx.container.get(WorkspacePreferenceProvider);
-        return [userProvider, workspaceProvider];
-    });
+    bind(PreferenceProvider).to(UserPreferenceProvider).inSingletonScope().whenTargetNamed(PreferenceScope.User);
```

Comment:
there is `rebind` to override a particular binding:
```ts
rebind(PreferenceProvider).to(UserPreferenceProvider).inSingletonScope().whenTargetNamed(PreferenceScope.User);
```

the same for the workspace

Reply:
It didn't work for me
it throws `Error: No matching bindings found for serviceIdentifier: PreferenceProvider`

Reply:
ok, let's go with `unbind` for now, although it will remove all bindings :( I've opened an issue https://github.com/inversify/InversifyJS/issues/853